river/ensemble/streaming_random_patches.py

```diff
@@ -486,85 +634,457 @@ def predict_proba_one(self, x):
 
         return self.model.predict_proba_one(x_subset)
 
-    def _trigger_warning(
-        self, all_features, n_samples_seen: int, rng: np.random.Generator
+    def predict_one(self, x: dict) -> base.typing.ClfTarget:
```

Comment:
I intentionally did not make it an extension of `base.Classifier` to avoid confusion with a 'normal' classifier. This base class is specific to SRP, the `learn_one` method has a different signature.

Returning `None` or `0` would result in the same mismatch for a string. However, we might be lucky if there is a class `0` ;-)